plugin.video.themoviedb.helper/resources/lib/container.py

```diff
@@ -297,97 +263,316 @@ def list_traktuserlists(self):
+
+        (limit, rnd_list) = (50, 10) if self.params.pop('random', False) else (10, None)
+
+        self.list_items(
+            items=traktapi.get_itemlist(
+                cat.get('path', '').format(**params), page=self.params.get('page', 1), limit=limit, rnd_list=rnd_list, req_auth=cat.get('req_auth'),
+                key_list=['movie', 'show'] if self.params.get('type') == 'both' else [utils.type_convert(self.item_tmdbtype, 'trakt')]),
+            url={'info': cat.get('url_info', 'details')})
+
+    def list_traktmanagement(self):
+        if not self.params.get('trakt') in constants.TRAKT_MANAGEMENT:
+            return
+        with utils.busy_dialog():
+            traktapi = traktAPI()
+            slug_type = 'show' if self.params.get('type') == 'episode' else utils.type_convert(self.params.get('type'), 'trakt')
+            trakt_type = utils.type_convert(self.params.get('type'), 'trakt')
+            slug = traktapi.get_traktslug(slug_type, 'tmdb', self.params.get('tmdb_id'))
+            item = traktapi.get_details(slug_type, slug, season=self.params.get('season', None), episode=self.params.get('episode', None))
+            items = {trakt_type + 's': [item]}
+            if self.params.get('trakt') == 'watchlist_add':
+                traktapi.sync_watchlist(slug_type, mode='add', items=items)
+            if self.params.get('trakt') == 'history_add':
+                traktapi.sync_history(slug_type, mode='add', items=items)
+            if self.params.get('trakt') == 'collection_add':
+                traktapi.sync_collection(slug_type, mode='add', items=items)
+            if self.params.get('trakt') == 'watchlist_remove':
+                traktapi.sync_watchlist(slug_type, mode='remove', items=items)
+            if self.params.get('trakt') == 'history_remove':
+                traktapi.sync_history(slug_type, mode='remove', items=items)
+            if self.params.get('trakt') == 'collection_remove':
+                traktapi.sync_collection(slug_type, mode='remove', items=items)
+            # TODO: Check status response and add dialog
+        self.updatelisting = True
+
+    def list_becauseyouwatched(self, mostwatched=False):
+        traktapi = traktAPI(tmdb=self.tmdb, login=True)
+        userslug = traktapi.get_usernameslug()
+        if not userslug:
+            return
+        func = traktapi.get_mostwatched if mostwatched else traktapi.get_recentlywatched
+        recentitems = func(userslug, self.params.get('type'), limit=5, islistitem=False)
+        recentitem = recentitems[random.randint(0, len(recentitems) - 1)]
+        if not recentitem[1]:
+            return
+        self.plugincategory = recentitem[2]
+        self.params['tmdb_id'] = recentitem[1]
+        self.params['info'] = 'recommendations'
+        self.list_tmdb()
+
+    def list_getid(self):
+        self.params['tmdb_id'] = self.get_tmdb_id(**self.params)
+
+    def list_play(self):
```

Comment:
I assume this is only called when playing a playlist? Our api should be improved in this regard

Reply:
It's a badly named function - it's a call back play function for listitems. The Player module takes tmdb_id, looks-up details online and the compares against local db. If a matching item is found in local db, it is played. 

If a local item can't be found, plugin presents user with options to play or search with an external plugin. The actions to play with an external plugin are defined in player.json files which are placed in `userdata/addon_data/plugin.video.themoviedb.helper/players/`